torch/csrc/utils/python_arg_parser.h

```diff
@@ -156,7 +156,7 @@ inline std::vector<at::Tensor> PythonArgs::tensorlist(int i) {
 }
 
 inline std::vector<int64_t> PythonArgs::intlist(int i) {
-  if (!args[i]) return signature.params[i].default_intlist;
+  if (!args[i] || args[i] == Py_None) return signature.params[i].default_intlist;
```

Comment:
Do we really want to treat None as “return the default”? I don’t think so

Reply:
What else do you want it to be? Or are you saying that `args[i]` should never be `None`?

Reply:
yes, it should never be None. `NULL` means that someone didn't specify it, so you should take the default value. `None` means that it **was specified**, but it was `None` and not a list. This is a type mismatch

Reply:
Here's the prototype for avg_pool3d:

```
avg_pool3d(Tensor input, IntList[3] kernel_size, IntList[3] stride=None, IntList[3] padding=0, bool ceil_mode=False, bool count_include_pad=False)
```

Here's the desired behavior:

`kernel_size`: required and should not allow explicit `None` (current behavior)
`stride`: optional and should not treat explicit `None` as an empty list (current behavior)
`padding`: optional. I'm ambivalent about treating `None` as the default here, which is the current behavior.

This change was to fix the `stride` case, which is important because it lets you write Python functions with the same defaults that pass the full argument list down to C++.

Reply:
Ooops had a big typo. If your reading this from an email, please look at the GItHub comments instead.

Reply:
The reason I inserted a None test here was because I was getting a None from Python, and this case was not being checked otherwise. I'm OK with whatever semantics we ascribe for None, but let's make sure we are actually doing all the type tests we need in this code.

Reply:
I'm ok with accepting this for now to unbreak ONNX quickly, and I understand the case of `stride`, but it still would seem clearer for me if we didn't accept `None` for arbitrary arguments that have defaults